test/run_test.py

```diff
@@ -340,6 +349,82 @@ def print_to_stderr(message):
     print(message, file=sys.stderr)
 
 
+def get_test_time_reports_from_S3():
+    commit_date_str = subprocess.check_output(
+        ['git', 'show', '-s', '--format=%ci', 'HEAD'],
+        encoding="ascii").splitlines()[0]
+    commit_date = datetime.strptime(commit_date_str, '%Y-%m-%d %H:%M:%S %z')
+    day_before_commit = str(commit_date - timedelta(days=1)).split(' ')[0]
+    # something like git rev-list --before="2021-02-24" --max-count=1 --remotes="*upstream/nightly"
+    nightly_commit = subprocess.check_output(
+        ["git", "rev-list", f"--before={day_before_commit}", "--max-count=1", "--remotes=*upstream/nightly"],
+        encoding="ascii").splitlines()[0]
+
+    job = os.environ.get("CIRCLE_JOB", "")
+    job_minus_shard_number = job.rstrip('0123456789')
+    s3 = boto3.resource("s3", config=botocore.config.Config(signature_version=botocore.UNSIGNED))
+    bucket = s3.Bucket(name="ossci-metrics")
+    summaries = bucket.objects.filter(Prefix=f"test_time/{nightly_commit}/{job_minus_shard_number}")
+
+    reports = []
+    for summary in summaries:
+        binary = summary.get()["Body"].read()
+        string = bz2.decompress(binary).decode("utf-8")
+        reports.append(json.loads(string))
+    return reports
+
+
+def calculate_job_times(reports) -> Dict[str, Tuple[float, int]]:
+    # an entry will be like ("test_file_name" -> (current_avg, # values))
+    jobs_to_times: Dict[str, Tuple[float, int]] = dict()
+    for report in reports:
+        assert report['format_version'] == 2, "S3 format currently handled is version 2 only"
```

Comment:
not sure if this matters since the point is that the `assert` just raises an error anyways, but maybe we should use `.get` to handle the case where the key is absent?
```suggestion
        assert report.get('format_version') == 2, "S3 format currently handled is version 2 only"
```